### DISCUSSION TITLE
Verification not with realtime only with a picture

### DISCUSSION THREAD

**Discussion by hrojasbtk** [shubham0204/OnDevice-Face-Recognition-Android#13] [General]
Hi,
Please I have a question, I´m develop a new application using your project as example but I dont need to use camera stream frame by frame, is possible to add in this project a new activity o similar that can take picture, send that picture to face verfication process and get result of verification. Or maybe there are any considerations that I have to know.

**Comment by shubham0204** [comment]
You can start by adding the [FaceNet.kt](https://github.com/shubham0204/OnDevice-Face-Recognition-Android/blob/main/app/src/main/java/com/ml/shubham0204/facenet_android/domain/embeddings/FaceNet.kt) class to your project. The `getFaceEmbedding` method requires a cropped face image, for which we can use [Mediapipe's Face Detector](https://ai.google.dev/edge/mediapipe/solutions/vision/face_detector/android) or [MLKit's FaceDetection API](https://developers.google.com/ml-kit/vision/face-detection/android). Prior to face detection, we may have to perform image rotation considering the EXIF data stored in the selected image. See [this](https://gist.github.com/tomogoma/788e3b775dd611c9226f8e17781a0f0c) gist for the code.
